lib/network/rest/server.js

```diff
@@ -0,0 +1,332 @@
+'use strict'; // eslint-disable-line
+
+const httpServer = require('../http/server');
+const Logger = require('werelogs').Logger;
+const url = require('url');
+const assert = require('assert');
+
+const constants = require('../../constants');
+const utils = require('./utils');
+const errors = require('../../errors');
+
+/**
+ * Parse the Range header into a two-integers array with possibly
+ * undefined values
+ *
+ * @param {String} rangeHeader - The 'Range' header value
+ * @return {(Number|undefined) []} An array of [left, right] integer
+ *   values provided in the range specification if present, or
+ *   undefined otherwise.
+ */
+function parseRangeSpec(rangeHeader) {
```

Comment:
To be honest I'm not very satisfied with the current implementation of the S3 function (but this is probably my picky nature :p) hence my reluctance to use it as-is. I find it a bit too complex for what it is for, as it does not decouple the parsing concern from the processing, and the parsing could be both more robust and simpler if it was using a regexp (unless there's a performance concern about regexp, though when compiled they should be rather fast).

See e.g.:
```
    const rangePortion = rangeHeader.replace('bytes=', '').split('-');
    if (rangePortion.length > 2) {
        return { range };
    }
```

This would not prevent e.g. `bytes=bytes=5-10bytes=` to be processed as a valid range. Of course it's not likely to happen unless someone forges a request like that, so it's nothing critical, just noticed it on the way.

```
    if (start >= totalLength && end >= totalLength) {
        return { range, error: errors.InvalidRange };
    }
```
This may be the answer to my unrelated question about how we handle errors that are returned and not thrown.